src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-list/pool-list.component.ts

```diff
@@ -232,6 +234,19 @@ export class PoolListComponent extends ListWithDetails implements OnInit {
     };
   }
 
+  fetchErasureCodeProfile(erasureCodeProfile: string) {
+    this.ecpService.list().subscribe((ecProfileList: ErasureCodeProfile[]) => {
```

Comment:
I think caching this in the front-end is a good idea. Additionally we should add a comment to the API endpoint that this is an expensive operation and the consumer of this endpoint should consider caching this. I don't know if it is a good idea to cache this on the back-end because as a consumer i expect the lastest data when calling the endpoint. But maybe this is a personal point of view. If it is a common approach to cache such things on the back-end side i'm also fine with this.

Reply:
Perhaps it's a bad trait of mine, but I tend to prefer 'defensive coding' 🙈 It's not about mistrust, bad intentions or incompetence, but because in heavily layered (and encapsulated) systems like the dashboard, when you are coding at the top of the stack, you shouldn't care too much about the performance at the bottom of it. In the end, nobody knows better about the internals of component than the component itself...

That doesn't mean that a back-end component has to cheat its consumers: if we provide a cached response we could deliver it along with the timestamp or a uuid of the response, so the consumer can know how fresh/stale the response is.